### ISSUE TITLE
Pressing spacebar while viewing a palette's contextual menu deletes the palette

### ISSUE THREAD

**Issue by Riley-Sullivan** [musescore/MuseScore#22025]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Steps to reproduce

1. Open any score or make a new one.
2. Right-click a palette to bring up that palette's contextual menu.
3. Press the spacebar on your computer keyboard.
4. See error: the palette has been deleted.

### MuseScore Version

4.2.1

### Regression

I don't know

### Operating system

macOS 12.7.2

### Additional context

It also happens if you open the palette via the "..." button—not just right-click.
Also, the same thing happens to items *inside* a palette—if you right-click them and press the spacebar, they will be deleted.
Then the only way I can see to get everything back is to force-quit MuseScore so that layout preferences are not saved.

**Comment by MarcSabatella** [comment]
I can confirm the bug, but I just wanted to adding, the workaround would be to click "Add palettes" to re-add the deleted palette.  Deleted default palettes are never fully removed and are always available to re-add.  And if it *isn't* a default palette, you'd get a confirmation dialog on the attempt to delete.

**Comment by Riley-Sullivan** [comment]
On further investigation, I discovered that the spacebar always chooses the first (top) item in any contextual menu, app-wide. For example, right-click a measure and spacebar will choose the Cut option, which is at the top of the contextual menu.

Maybe the title of this issue should be changed to reflect that?

**Comment by wizofaus** [comment]
That's not true if you use the down arrow first to select a different option though. But it seems in general the palette is virtually impossible to navigate/use with just the keyboard, unless I'm missing something.

**Comment by MarcSabatella** [comment]
No, it’s quite possible, as discussed on Discord.  Just a little non-intuitive as far ass how to enter a palette once selected.
